### PR TITLE
Restructuring from PRY towards PRX 

### PR DESCRIPTION
**Pull request by positr0nium**
## Description

Replace the PRY (Phased-RY) gate with a native PRX (Phased-RX) gate throughout Qrisp's pass management and Qiskit converter. PRX is the native single-qubit gate of IQM hardware and maps directly to Qiskit's `RGate`. This removes an unnecessary intermediate representation (PRY) that required error-prone parameter transformations.

## Related Issues

Closes #
Related to #

## Type of Change

- [x] Feature (new functionality)
- [ ] Change Request (modification of existing functionality)
- [ ] Bug Fix
- [x] Refactoring (no behavior change)
- [ ] Performance improvement
- [ ] Documentation
- [ ] CI / Build

## Breaking Change?

- [x] Yes
- [ ] No

If yes, describe the impact and migration path:

The `PRYGate` class and `convert_to_pry` pass have been removed. Any code importing `PRYGate` or `convert_to_pry` from `qrisp.circuit` must be updated to use `PRXGate` and `convert_to_prx` instead. The parameter relationship is: `PRX(alpha, beta)` is equivalent to `PRY(alpha, beta - pi/2)`.

## What was changed?

- **Added `PRXGate`** — subclass of `U3Gate` representing the gate `RZ(beta) · RX(alpha) · RZ(-beta)`. Constructor accepts `alpha` (rotation angle) and `beta` (phase). Includes an `inverse()` method returning `PRXGate(-alpha, beta)`.
- **Added `convert_to_prx` pass** — converts arbitrary U3 single-qubit gates to PRX gates. Handles the single-PRX case (`lambda ≈ -phi`) and the two-PRX decomposition. Accumulates global phase across decompositions and emits a trailing `GPhaseGate` on qubit 0 to preserve exact unitary equivalence.
- **Removed `PRYGate` and `convert_to_pry`** — the intermediate PRY representation is eliminated. File `convert_to_pry.py` deleted.
- **Added `collect_gphases` to `decompose` pass** — new `collect_gphases: bool = False` parameter. When enabled, all `GPhaseGate` instructions in the decomposed circuit are stripped, their phases accumulated, and a single `gphase` gate emitted on the first qubit of the last instruction with `num_qubits > 0` (avoids anchoring on freshly allocated qubits). Used internally by the compilation pipeline.
- **Qiskit converter** — `PRXGate` maps directly to Qiskit's `RGate(alpha, beta)` in both directions with no parameter transformation.
- **Documentation** — added `convert_to_prx.rst` and updated the built-in passes index.
- **Tests** — new unit tests for `PRXGate` class, `convert_to_prx` pass, and `collect_gphases` feature. Added PRX round-trip tests and 36 direct unitary comparison tests (Qrisp ↔ Qiskit, no round-trip) that handle the MSB-first vs LSB-first qubit ordering difference between the two frameworks.

## How was it tested?

| Test-ID | Status |
|---------|--------|
| T-001 `test_PRXGate.py` — construction, inheritance, name, params, inverse, unitary correctness | ✅ |
| T-002 `test_convert_to_prx.py` — single/two PRX cases, CX preservation, empty circuit, global phase tracking, 50 random circuit fidelity | ✅ |
| T-003 `test_qiskit_converter.py` — PRX round-trip, RGate→PRX, controlled PRX, 36 direct unitary comparisons, 90 total | ✅ |
| T-004 `test_decompose.py` — 6 collect_gphases tests (merging, anchoring, unitary preservation, edge cases), 24 total | ✅ |

## Screenshots / Output (if applicable)

```
tests/interface_tests/test_qiskit_converter.py: 90 passed in 0.36s
tests/circuit_tests/passes/test_PRXGate.py: All PRXGate tests passed!
tests/circuit_tests/passes/test_convert_to_prx.py: All convert_to_prx tests passed!
tests/circuit_tests/passes/test_decompose.py: 24 passed (6 collect_gphases)
```

## Checklist
- [x] My code follows the project's coding standards
- [x] I have performed a self-review
- [x] I have added/updated tests (referencing issue Test-IDs)
- [x] All tests pass locally and in CI
- [x] I have updated the documentation
- [ ] I have added a changelog entry
- [x] Breaking changes are documented with migration path

## Reviewer Notes

- The `PRXGate.alpha` and `PRXGate.beta` attributes store the native PRX parameters. The inherited `theta`/`phi`/`lam` attributes reflect the U3 parameterization: `U3(alpha, beta - pi/2, pi/2 - beta)`.
- The `convert_to_prx` pass includes global phase tracking via a trace-based phase extraction and emits a `GPhaseGate` at the end of the circuit to preserve exact unitary equivalence — not just up to global phase.
- The new `collect_gphases` parameter on `decompose` is off by default. When enabled, it merges scattered gphase gates into a single one anchored on the last non-trivial instruction's qubit (avoiding qubit-freshness issues). The compilation pipeline uses `collect_gphases=True` internally.
- The Qiskit converter's `sx` gate mapping was found to have a pre-existing global phase discrepancy between Qrisp's and Qiskit's SXGate conventions. This is noted but not addressed in this PR.

### CODE REVIEW

File: tests/circuit_tests/passes/test_convert_to_prx.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+from qrisp import QuantumCircuit, Qubit, U3Gate, convert_to_prx
+from qrisp.circuit import PRXGate
+import numpy as np
+
+
+def test_convert_to_prx():
+    """Test the convert_to_prx function with various quantum circuits."""
+
+    print("Testing convert_to_prx...")
```
Comment (renezander90) [inline tests/circuit_tests/passes/test_convert_to_prx.py]: These print statements in `test_PRXGate.py` and `test_convert_to_prx.py` should be removed. Compared to the other tests in this folder, both files lack structure in terms of classes/functions for specific tests (see e.g., `test_convert_to_cx.py`).

**Comment by renezander90** [comment]
> The Qiskit converter's sx gate mapping was found to have a pre-existing global phase discrepancy between Qrisp's and Qiskit's SXGate conventions. This is noted but not addressed in this PR.

Is `sx` converted correctly to qiskit or is there a bug? If so, an issue should be created.

**Review by renezander90 — APPROVED** [review]
🎊
